src/config/mod.rs

```diff
@@ -91,12 +91,22 @@ impl<'c> Config<'c> {
     }
 
     /// Create boot source from `self`.
-    pub fn boot_source(&self) -> BootSource<'_> {
-        BootSource {
-            kernel_image_path: self.kernel_image_path.as_ref(),
-            initrd_path: self.initrd_path.as_ref().map(AsRef::as_ref),
+    pub fn boot_source(&self) -> Result<BootSource<'_>, Error> {
+        let initrd_filename = match &self.initrd_path {
+            Some(initrd_path) => {
+                let initrd_filename = initrd_path.file_name().ok_or(Error::InvalidInitrdPath)?;
+                Some(Path::new(initrd_filename))
+            }
+            None => None,
+        };
+
+        let kernel_image_file = Path::new(KERNEL_IMAGE_FILENAME);
+
+        Ok(BootSource {
+            kernel_image_path: kernel_image_file,
+            initrd_path: initrd_filename,
```

Comment:
if these needs to be filenames, then the fields need renaming.

Reply:
The names are part of socket API

Reply:
ah right, i forgot again we're serializing the config struct. :facepalm: 

Let's make sure these need to be filenames.